### ISSUE TITLE
Oversees wrong declaration roder

### ISSUE THREAD

**Issue by axkibe** [eslint/eslint#11901]
* **ESLint Version:** v5.16.0
* **Node Version:** v11.4.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**  default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    "env": {
        "browser": true,
        "es6": true,
    },
    "extends": "eslint:recommended",
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.** I did ```bash eslint test.js ```

```js
window.onload = function( )
{
	const x = 3 + y;
	const y = 0;
	return x;
}
```

**What did you expect to happen?**

I expected it to complain like chrome does, that y is not yet definied. (This is a simplified version of something more complex in my code)

**What actually happened? Please include the actual, raw output from ESLint.**

Eslint didn't see anything wrong with above code.

**Are you willing to submit a pull request to fix this bug?**

Sorry, No :)

**Comment by aladdin-add** [comment]
thanks for the issue. it can be caught by https://eslint.org/docs/rules/no-use-before-define

I think we can add it in eslint-recommended (semver-major change).
